pkg/hintrunner/hintrunner.go

```diff
@@ -4,25 +4,215 @@ import (
 	"fmt"
 
 	VM "github.com/NethermindEth/cairo-vm-go/pkg/vm"
+	mem "github.com/NethermindEth/cairo-vm-go/pkg/vm/memory"
+	f "github.com/consensys/gnark-crypto/ecc/stark-curve/fp"
 )
 
-// todo: Can two or more hints be assigned to a specific PC?
+// Used to keep track of all dictionaries data
+type Dictionary struct {
+	// The data contained on a dictionary
+	data map[f.Element]*mem.MemoryValue
+	// Unique id assigned at the moment of creation
+	idx uint64
+}
+
+// Gets the memory value at certain key
+func (d *Dictionary) At(key *f.Element) (*mem.MemoryValue, error) {
+	if value, ok := d.data[*key]; ok {
+		return value, nil
+	}
+	return nil, fmt.Errorf("no value for key %s", key)
+}
+
+// Given a key and a value, it sets the value at the given key
+func (d *Dictionary) Set(key *f.Element, value *mem.MemoryValue) {
+	d.data[*key] = value
+}
+
+// Returns the initialization number when the dictionary was created
+func (d *Dictionary) InitNumber() uint64 {
+	return d.idx
+}
+
+// Used to manage dictionaries creation
+type DictionaryManager struct {
+	// a map that links a segment index to a dictionary
+	dictionaries map[uint64]Dictionary
+}
+
+func InitializeDictionaryManagerIfNot(ctx *HintRunnerContext) {
+	if ctx.DictionaryManager.dictionaries == nil {
+		ctx.DictionaryManager.dictionaries = make(map[uint64]Dictionary)
+	}
+}
+
+// It creates a new segment which will hold dictionary values. It links this
+// segment with the current dictionary and returns the address that points
+// to the start of this segment
+func (dm *DictionaryManager) NewDictionary(vm *VM.VirtualMachine) mem.MemoryAddress {
+	newDictAddr := vm.Memory.AllocateEmptySegment()
+	dm.dictionaries[newDictAddr.SegmentIndex] = Dictionary{
+		data: make(map[f.Element]*mem.MemoryValue),
+		idx:  uint64(len(dm.dictionaries)),
+	}
+	return newDictAddr
+}
+
+// Given a memory address, it looks for the right dictionary using the segment index. If no
+// segment is associated with the given segment index, it errors
+func (dm *DictionaryManager) GetDictionary(dictAddr *mem.MemoryAddress) (Dictionary, error) {
+	dict, ok := dm.dictionaries[dictAddr.SegmentIndex]
+	if !ok {
```

Comment:
nit: All other functions in this file follow pattern 
```
if dict, ok := ...; ok{
     return value;
}
return error;
```
this one is the only that does the inverse